### PR TITLE
Extend test beyond unbondng time + fixes

### PR DESCRIPTION
**Pull request by joneskm**

### CODE REVIEW

File: x/staking/src/keeper/unbonding.rs [hunk 1]
```diff
@@ -443,7 +441,7 @@ impl<
     ) -> Result<(), GasStoreErrors> {
         let store = TransactionalContext::kv_store_mut(ctx, &self.store_key);
         let mut store = store.prefix_store_mut(VALIDATOR_QUEUE_KEY);
-        store.delete(&validator_queue_key(&end_time, end_height))?;
+        store.delete(&validator_queue_key(&end_time, end_height))?; //TODO: should we NOT be using the prefix store here - since the validator_queue_key adds the prefix?
```
Comment (NYBACHOK) [inline x/staking/src/keeper/unbonding.rs:444]: What about app hash? If it correct then no, but in other case we should omit prefix store,

File: x/staking/src/keys.rs [hunk 2]
```diff
@@ -54,11 +54,9 @@ pub(super) fn parse_validator_queue_key(key: &[u8]) -> anyhow::Result<(Timestamp
     Ok((time, height))
 }
 
+// This is the key for use in the unbonding queue sub store (UNBONDING_QUEUE_KEY prefix)
 pub(super) fn unbonding_delegation_time_key(time: &Timestamp) -> Vec<u8> {
-    let tbz = time.format_bytes_rounded();
-    let mut bz = UNBONDING_QUEUE_KEY.to_vec();
-    bz.extend(tbz.iter());
-    bz
+    time.format_bytes_rounded()
```
Comment (NYBACHOK) [inline x/staking/src/keys.rs:59]: Maybe it better to remove such method completely?

**Review by NYBACHOK — APPROVED** [review]

File: x/staking/src/keeper/unbonding.rs (hunk 1 above)
Comment (joneskm) [reply x/staking/src/keeper/unbonding.rs:444]: I need to check whether we have a test which takes this path. If we do then I'll remove the TODO, if we don't then I'll add a test.

File: x/staking/src/keys.rs (hunk 2 above)
Comment (joneskm) [reply x/staking/src/keys.rs:59]: I've kept it in for now...

**Pull request merged**